recipes/react-router/app/routes/puck-splat.tsx

```diff
@@ -0,0 +1,93 @@
+import { useFetcher, useLoaderData } from "react-router";
+import type { Data } from "@measured/puck";
+import { Puck, Render } from "@measured/puck";
+
+import "@measured/puck/puck.css";
+
+import type { Route } from "./+types/puck-splat";
+import { config } from "~/puck.config";
+import { resolvePuckPath } from "~/lib/resolve-puck-path.server";
+import { getPage, savePage } from "~/lib/pages.server";
+
+export async function loader({ params }: Route.LoaderArgs) {
+  const pathname = params["*"];
+  const { isEditorRoute, path } = resolvePuckPath(pathname);
+  let page = await getPage(path);
+
+  // Throw a 404 if we're not rendering the editor and data for the page does not exist
+  if (!isEditorRoute && !page) {
+    throw new Response("Not Found", { status: 404 });
+  }
+
+  // Empty shell for new pages
+  if (isEditorRoute && !page) {
+    page = {
+      content: [],
+      root: {
+        props: {
+          title: "",
+        },
+      },
+    };
+  }
+
+  return {
+    isEditorRoute,
+    path,
+    data: page,
+  };
+}
+
+export function meta({ data: loaderData }: Route.MetaArgs) {
+  return [
+    {
+      title: loaderData.isEditorRoute
+        ? `Edit: ${loaderData.path}`
+        : loaderData.data.root.title,
+    },
+  ];
+}
+
+export default function PuckSplatRoute({ loaderData }: Route.ComponentProps) {
+  return (
+    <div>
+      {loaderData.isEditorRoute ? (
+        <Editor />
+      ) : (
+        <Render config={config} data={loaderData.data} />
+      )}
+    </div>
+  );
+}
```

Comment:
Might make sense to move this to the bottom?

Reply:
I was following the convention used in route files for RR/Remix but I can update this